app/models/bus_stop.rb

```diff
@@ -251,7 +251,8 @@ def name_with_id
   end
 
   def self.strip_id_from_name(name_with_id)
-    name_with_id[/[^(]+/].strip
+    id_string = name_with_id.match(/ \(\d+\)$/)[0]
```

Comment:
Why the intermediate step?

```
name_with_id.gsub(/ \(\d+\)$/ , '')
```

Reply:
I'm interested to know too. This was a creative way to do it.

Reply:
I wrote it that way while trying to decode what the original was doing - for the sake of communicativeness I left it with the intermediate variable.

I can condense it if there's a strong preference.